packages/hotspot/src/hotspot/index.jsx

```diff
@@ -2,17 +2,48 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import Typography from '@material-ui/core/Typography';
 import {Collapsible, hasText, PreviewPrompt} from '@pie-lib/render-ui';
-import Container from './container';
 import { withStyles } from '@material-ui/core/styles';
 
+import Container from './container';
+
 class HotspotComponent extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      showCorrect: false
+      showCorrect: false,
+      observer: null,
+      scale: 1
     };
   }
 
+  componentDidMount() {
+    this.observer = new MutationObserver((mutations) => {
+      mutations.forEach(() => {
+        const target = document.getElementById('question-container')?.style?.cssText
+        const zoom = target?.substring(
+          target.indexOf('--pie-zoom') + 11,
+          target.lastIndexOf('%')
+        );
+
+        const zoomParsed = zoom?.replace(/\s/g, '');
+        if (zoomParsed) {
+          this.setState({
```

Comment:
Do you think it would make sense to check if the value is different before updating the state?
Just to make sure we don't cal setState if it's not necessary?